provider/datagen/Cargo.toml

```diff
@@ -77,7 +77,7 @@ tinystr = { version = "0.7.1", path = "../../utils/tinystr", features = ["alloc"
 toml = "0.5"
 writeable = { version = "0.5.1", path = "../../utils/writeable" }
 zerovec = { version = "0.9.2", path = "../../utils/zerovec", features = ["serde", "yoke"] }
-zip = { version = "0.6", default-features = false, features = ["deflate"] }
+zip = { version = ">=0.5, <0.7", default-features = false, features = ["deflate"] }
```

Comment:
you should also `cargo update -p zip` to get the new version so that the `time 0.1` dep disappears from the lockfile

Reply:
and cached-path as well

I'm fine maintaining the property that `time 0.1` isn't in our lockfile, even if clients will have the opportunity to use older crates.

Reply:
Sure but again the lock file does not effect the crate user, I thought it would be easier for u to keep it 0.5 while I can upgrade to 0.6 if that's preferable

Reply:
Yes, I understand how lockfiles work.

"A crate with a vulnerability is not in the lockfile" is easy for tooling to check and maintain, even though it doesn't mean we've completely excised the dep from all theoretical dependency chains.

And like I said clients needing to stick to zip 0.5 can, because we are giving them the flexibility.

Ultimately dependency management is something that the application developer at the end of the dependency tree must do: as library authors we can make the job _easy_ for them by reducing deps and keeping things flexible, but different people have different needs and we cannot guess those as library authors. Some people care about MSRV. Others have vendored crates and want to stick to those versions as much as possible. Different people have different security stances around vulnerabilities.

You're seeing the confluence of a bunch of different requirements here.

Reply:
@Manishearth time v0.1 is now removed from `Cargomlock`

Reply:
> And like I said clients needing to stick to zip 0.5 can, because we are giving them the flexibility.

@Manishearth  TBH I don't see why crate users need this flexibility for most of the cases (except sth like tokio_0, tokio_1) since the libs are statically linked. The dependency version should be transport to users and if any of the structs are used in public APIs they should be re-exported so that users do not need to care about it at all. If users need a different version of the same lib they could just use it with no problem (slightly bloating the binary but not a big deal), it will be problematic if the dependencies are dynamically linked (in java, or c#) but that's not the case in rust

Reply:
Because a lot of larger projects (for example Firefox, but there's also stuff in Google) vendor their dependencies, and also care about auditing dependencies. If they already have zip 0.5 they may prefer to stick to that. This is what I meant by "we're trying to keep dependencies in sync with other places": new dependencies are additional overhead for some people, and allowing users the flexibility to avoid that is good.

Not everyone is okay with a bloated dependency tree with multiple versions of the same crate. Different people have different needs and tradeoffs. Not every user of ICU4X is even using Cargo to drive the build.

Reply:
I think it's the best to use mono repo and workspace dependencies to avoid duplicate crates in dependency tree. For isolated crate repos it adds unnecessary and unmanageable maintenance burden. Anyways thanks for the review!

Reply:
........ yes, precisely, _that's my point_. We have clients that use monorepos. They would like to avoid duplicate crates, and they would like to avoid having to patch ICU4X's dep spec (or have to _figure out_ when our dependencies are malleable). Expressing that flexibility with range deps helps.

_We_ cannot use a dependency-vendored monorepo because ultimately we are a _library_ and have multiple clients. We aren't trying to avoid duplicate crates, but we are trying to ensure that we do not _force_ duplicate crates onto clients as much as possible.

This is a perfectly manageable maintenance burden for us. Aside from datagen, ICU4X has a low number of dependencies, and even datagen has a manageable number of dependencies if built in certain modes, and we have tests ensuring that.

Reply:
Arguably, as a lib crate I think the CI needs to cover all combinations of zip, cached-path versions. And if more crate more versions are added the num of combinations soon become 'unmanageable'

Reply:
As I said already, we do have plans for minimal/maximal version testing, and hope to do an MSRV test with minimal/maximal versions for a subset of the project (so, not datagen, would not affect zip/cached-path). Currently we achieve some of these properties to an acceptable level by having very few dependencies on the main part of the project.

There are diminishing returns after you have minimal/maximal testing.

And ultimately it is normal in the Rust community for applications to run a bunch of `cargo update --precise`s if they have specific additional constraints for their dependencies.